File: 数学/168.cpp
#include<string>
#include<iostream>
#include<algorithm>
using namespace std;
class Solution {
public:
    string convertToTitle(int columnNumber) {
        string ans;
        int mod;
        while(columnNumber>0){
            mod = columnNumber%26;
            if(mod){
                ans.push_back('A'+(mod-1));
                columnNumber/=26;
            }
            else{
                ans.push_back('Z');
                columnNumber/=26;
                columnNumber--;
            }
        }
        reverse(ans.begin(),ans.end());
        return ans;
    }
};

int main()
{
    int columnNumber=52;
    Solution obj;
    cout<<obj.convertToTitle(columnNumber)<<endl;
}

File: 数学/168.py
class Solution:
    def convertToTitle(self, columnNumber: int) -> str:
        dic = {}
        for i in range(1,27):
            dic[i] = chr(ord("A")+(i-1))

        res = []
        while(columnNumber):
            mod = columnNumber%26
            if mod:
                res.append(dic[mod])
                columnNumber//=26
            else:
                res.append("Z")
                columnNumber//=26
                columnNumber-=1
        return "".join(res[::-1])

obj = Solution()
columnNumber=30
print(obj.convertToTitle(columnNumber))